src/identity/manager.rs

```diff
@@ -32,16 +36,68 @@ impl fmt::Display for Identity {
 #[derive(Clone)]
 pub struct SecretManager {
     client: CaClient,
+    cache: Arc<Mutex<HashMap<Identity, tls::Certs>>>,
 }
 
 impl SecretManager {
     pub fn new(cfg: crate::config::Config) -> SecretManager {
         let client = CaClient::new(cfg.auth);
-        SecretManager { client }
+        let cache: HashMap<Identity, tls::Certs> = Default::default();
+        return SecretManager { client: client, cache: Arc::new(Mutex::new(cache)) };
+    }
+
+    pub async fn refresh_handler(id: Identity, ctx: SecretManager, initial_sleep_time: u64) {
+        info!("refreshing certs for id {} in {:?} seconds", id, initial_sleep_time);
+        sleep(Duration::from_secs(initial_sleep_time)).await;
+        loop {
+            match ctx.client.clone().fetch_certificate(id.clone()).await {
+                Err(e) => {
+                    // Cert refresh has failed. Drop cert from the cache.
+                    warn!("Failed cert refresh for id {:?}: {:?}", id, e);
+                    { // lock cache
```

Comment:
do we need {} ?

Reply:
It is used to scope the lock.  Can replace with an explicit drop, but I personally like the visual indicator of where the lock is held.  Either way.